service/history/decision/task_handler.go

```diff
@@ -789,6 +806,19 @@ func (handler *taskHandlerImpl) handleDecisionContinueAsNewWorkflow(
 		return nil
 	}
 
+	if is, _ := handler.mutableState.IsCancelRequested(); is {
+		// cancellation must be sticky, as it's telling things to stop.
+		// this is particularly important for child workflows, as if they restart themselves after the parent
+		// cancels its context, there is no way for the parent to cancel the new run.
+		cancelAttrs := types.CancelWorkflowExecutionDecisionAttributes{
+			Details: nil, // TODO: serialize continue-as-new data somehow, may deserve a new field
```

Comment:
No need to get fancy, maybe even just a string here or something might be nice to indicate what has occurred

Reply:
I was considering that, yeah.  think it's worth just putting in a json string for now?  `"ContinueAsNew ignored because workflow is canceled"`?

Reply:
For now, I think I'm going to leave this empty.  Nil is a bit "safer" since we'll be storing structured data here in the future.

If we get enough confusion around this behavior though, even just hardcoding an explanation may be fine.  
When we get an "underlying reason"-like field added, that can be used to differentiate the type of data stored, which will be necessary anyway - a CanceledError could store a ContinueAsNew-details or a FailedError-details or possibly other things too.  An empty "underlying reason" seems like a fairly safe way to say "this is old data and cannot be deserialized that way".